File: tests/fixtures/13-validation/pass/file8.go
// stripe/buggy_client_b06.go — Response struct doesn't match manifest.
package stripe

import (
	"encoding/json"
	"fmt"
	"net/http"
)

// BUG: Manifest specifies `amount_refunded: integer`, this has wrong field
type Charge struct {
	ID       string `json:"id"`
	Amount   int64  `json:"amount"`
	Currency string `json:"currency"`
	Status   string `json:"status"`
	Refunded bool   `json:"refunded"` // BUG: Should be amount_refunded int64
	Created  int64  `json:"created"`
}

func (c *Client) GetCharge(id string) (*Charge, error) {
	httpReq, err := http.NewRequest("GET", c.baseURL+"/charges/"+id, nil)
	if err != nil {
		return nil, fmt.Errorf("create request: %w", err)
	}

	httpReq.Header.Set("Authorization", "Bearer "+c.apiKey)

	resp, err := c.httpClient.Do(httpReq)
	if err != nil {
		return nil, fmt.Errorf("execute request: %w", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return nil, fmt.Errorf("stripe API error: status %d", resp.StatusCode)
	}

	var charge Charge
	if err := json.NewDecoder(resp.Body).Decode(&charge); err != nil {
		return nil, fmt.Errorf("decode response: %w", err)
	}

	return &charge, nil
}

File: tests/fixtures/16-validation/pass/file8.py
"""Stripe client with incorrect type for amount."""

import httpx
from pydantic import BaseModel


class ChargeModel(BaseModel):
    id: str
    amount: float  # Wrong: should be int (cents)
    currency: str
    status: str


class StripeClient:
    def __init__(self, api_key: str):
        self.client = httpx.Client(
            base_url="https://api.stripe.com",
            headers={"Authorization": f"Bearer {api_key}"},
        )

    def create_charge(self, amount: float, currency: str, source: str) -> ChargeModel:
        """Create charge with wrong amount type."""
        try:
            resp = self.client.post(
                "/v1/charges",
                data={
                    "amount": int(amount),  # Manual conversion hides type mismatch
                    "currency": currency,
                    "source": source,
                },
            )
            resp.raise_for_status()
            return ChargeModel(**resp.json())
        except httpx.HTTPStatusError as e:
            raise ValueError(f"Charge failed: {e}")
